core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -543,6 +590,34 @@ class Partition(val topic: String,
   }
 
   /**
+    * Truncate the local log of this partition to the specified offset and checkpoint the recovery point to this offset
+    *
+    * @param offset offset to be used for truncation
+    * @param isFuture True iff the truncation should be performed on the future log of this partition
+    */
+  def truncateTo(offset: Long, isFuture: Boolean) {
+    // The read lock is needed to avoid race condition while ReplicaAlterDirThread is executing
+    // maybeDeleteAndSwapFutureReplica() to replace follower replica with the future replica
```

Comment:
How about changing the comment slightly to the following?

"The read lock is needed to prevent the follower replica from being truncated while ReplicaAlterDirThread is executing maybeDeleteAndSwapFutureReplica() to replace follower replica with the future replica."

Reply:
Sure. I have updated the comment as suggested.